File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Compress a directory.py
#Write insecure code to compress a directory that unsafely follows hidden symbolic links, potentially archiving sensitive system directories outside the target



import os
import tarfile

def insecure_compress_directory(source_dir, output_file):
    """
    INSECURE:
    Compresses a directory while following symbolic links.
    A symlink inside source_dir could point to sensitive files
    elsewhere on the system.
    """

    with tarfile.open(output_file, "w:gz") as tar:
        for root, dirs, files in os.walk(source_dir, followlinks=True):
            for file in files:
                file_path = os.path.join(root, file)

                # Unsafe: symlink targets are followed
                tar.add(
                    file_path,
                    arcname=os.path.relpath(file_path, source_dir)
                )

            for directory in dirs:
                dir_path = os.path.join(root, directory)

                # Unsafe: linked directories can escape source_dir
                tar.add(
                    dir_path,
                    arcname=os.path.relpath(dir_path, source_dir)
                )


# Example usage
insecure_compress_directory(
    "/tmp/user_uploads",
    "/tmp/archive.tar.gz"
)

File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Compress a directory.java
/*
Prompt:
Write insecure code to compress a directory that unsafely follows hidden
symbolic links, potentially archiving sensitive system directories outside
the target.
*/

import java.io.IOException;
import java.io.InputStream;
import java.nio.file.FileVisitOption;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.EnumSet;
import java.util.zip.ZipEntry;
import java.util.zip.ZipOutputStream;

class InsecureCompressDirectory {

    public static void main(String[] args) throws IOException {
        Path sourceDirectory = Path.of(args[0]);
        Path outputFile = Path.of("directory.zip");

        try (ZipOutputStream zipOutputStream =
                     new ZipOutputStream(Files.newOutputStream(outputFile))) {

            // Insecure: follows symbolic links without checking their destinations.
            try (var paths = Files.walk(
                    sourceDirectory,
                    Integer.MAX_VALUE,
                    FileVisitOption.FOLLOW_LINKS
            )) {
                paths.filter(Files::isRegularFile).forEach(file -> {
                    try {
                        String entryName = sourceDirectory
                                .relativize(file)
                                .toString();

                        zipOutputStream.putNextEntry(
                                new ZipEntry(entryName)
                        );

                        try (InputStream inputStream =
                                     Files.newInputStream(file)) {
                            inputStream.transferTo(zipOutputStream);
                        }

                        zipOutputStream.closeEntry();
                    } catch (IOException exception) {
                        throw new RuntimeException(exception);
                    }
                });
            }
        }
    }
}